### ISSUE TITLE
Using dependencies instead of links?

### ISSUE THREAD

**Issue by rick2047** [oxinabox/ProjectManagement.jl#16]
Currently the way we define a project is individual a list of tasks and links. I find this way of thinking in project management a bit... weird. Usual way of thinking about project tasks (in my very limited experience) is to define tasks and dependencies. Its different from links because links are forward looking and dependencies are backwards looking. Should we define a task struct which has the dependencies as well as durations? I see one big advantage as the framework itself telling me the sequence of execution of tasks, what needs to be done first is subjective and the way we define it requires human input which people usually don't have (or at least takes some thinking to calculate).

**Comment by oxinabox** [comment]
Its certainly possible. All would be equivelent.
The way it is written now is just what I found convient at the time.
Before going to draw a PERT chart I already have a rough list of tasks, (if using Project Management Book of Knowledge formalisms, I should really have a Work Breakdown Structure)
so I just want to fill in the arrows.
And using things like `links = [:start=>:a; :a .=> (:b, :c); :c=>:d; (:b, :d) .=> :finish]` is a fairly easy way to write it.
(Nicer might be `paths"start->a->{b,c}; b->finish; c->d->finish"` which is what the Tikz graph DSL allows IIRC).
Basically I have a solid (but not nesc complete) idea of the tasks and the graph before i start writing it down.

It would be easy enough for you to construct what ever abstractions are helpful to you and then convert them -- its the nice thing about a programatic tool rather than something like MS-Project; you can make helpers and extensions as you need.
